### PR TITLE
Updates to libensemble-rsmpi script

### PR DESCRIPTION
**Pull request by cchall**
Fixes for #90 and #100

### CODE REVIEW

File: rsopt/libe_tools/libensemble-rsmpi.py [hunk 1]
```diff
@@ -2,16 +2,17 @@
...
+run_string = 'rsmpi -n {n} -h {h} -t {t} {args}'
 
 parser = argparse.ArgumentParser(description="Wrap rsmpi for use by libensemble. Converts required MPICH flags"
                                              "to rsmpi flags.")
 
 parser.add_argument("-np", dest="n", required=True,
                     help="Pass number of processors to rsmpi")
-parser.add_argument("--ppn", dest="ppn", required=False,
-                    help="Not required by rsmpi, but supplied by libEnsemble. Input is not used if given.")
+parser.add_argument("--ppn", dest="t", required=False,
```
Comment (jeinstei) [inline rsopt/libe_tools/libensemble-rsmpi.py:14]: Is there any reason to have this default to a value? It is currently not `required` and will probably not act as expected if it is omitted

**Review by jeinstei — COMMENTED** [review]
It is behavior dependent how you want the `-t` argument handled if it is omitted

Everything looks fine otherwise

File: rsopt/libe_tools/libensemble-rsmpi.py [hunk 2]
```diff
@@ -27,21 +28,35 @@ def get_host_from_machinefile(machinefile):
     return host
 
 
+def stream_process(process):
+    go = process.poll() is None
+    for line in process.stdout:
```
[the remark this answers is not in the document]
Comment (cchall) [reply rsopt/libe_tools/libensemble-rsmpi.py:33]: That is a risk here. However, using `communicate` means that there is no way for the user to see stdout/stderr until the simulation completes.

File: rsopt/libe_tools/libensemble-rsmpi.py (hunk 1 above)
Comment (cchall) [reply rsopt/libe_tools/libensemble-rsmpi.py:15]: Good point. [8c93416](https://github.com/radiasoft/rsopt/pull/125/commits/8c93416d7121c4a6cb56f58f19f1450b147bfaba)

**Comment by cchall** [comment]
I'm going to go ahead and merge this. I've been using this for a few weeks now and it seems to be working well, and it is highly beneficial for monitoring jobs. There is no sign of deadlocks, but it is something keep in mind should issues arise with rsmpi execution.

**Pull request merged**